### DISCUSSION TITLE
[exporter/prometheus] Should an undefined `sum` be exported as 0?

### DISCUSSION THREAD

**Discussion by pichlermarc** [open-telemetry/opentelemetry-collector-contrib#12646] [Q&A]
As of https://github.com/open-telemetry/opentelemetry-collector/pull/5530 (update to proto v0.18), `sum` is now optional and accessing returns `0`. Therefore, if `sum` is not set, the prometheus exporter exports the sum as `0`.

I'm not very familiar with Prometheus and I'm trying to determine if I should open a bug issue or if that is intentional. :slightly_smiling_face:

**Comment by mx-psi** [comment]
cc @Aneurysm9 @dashpole

**Reply by dashpole** [reply]
An undefined sum should not be exported at all.  IIRC, https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9120 fixes this, but I need to rebase it as it got stale before it was reviewed.
